crates/turborepo-filewatch/src/lib.rs

```diff
@@ -89,53 +92,96 @@ pub struct FileSystemWatcher {
 }
 
 impl FileSystemWatcher {
-    pub async fn new_with_default_cookie_dir(
-        root: &AbsoluteSystemPath,
-    ) -> Result<Self, WatchError> {
+    pub fn new_with_default_cookie_dir(root: &AbsoluteSystemPath) -> Result<Self, WatchError> {
         // We already store logs in .turbo and recommend it be gitignore'd.
         // Watchman uses .git, but we can't guarantee that git is present _or_
         // that the turbo root is the same as the git root.
-        Self::new(root, &root.join_components(&[".turbo", "cookies"])).await
+        Self::new(root, root.join_components(&[".turbo", "cookies"]))
     }
 
-    pub async fn new(
+    pub fn new(
         root: &AbsoluteSystemPath,
-        cookie_dir: &AbsoluteSystemPath,
+        cookie_dir: AbsoluteSystemPathBuf,
     ) -> Result<Self, WatchError> {
-        if root.relation_to_path(cookie_dir) != PathRelation::Parent {
+        tracing::debug!("initing file-system watcher");
+
+        if root.relation_to_path(&cookie_dir) != PathRelation::Parent {
             return Err(WatchError::Setup(format!(
                 "Invalid cookie directory: {} does not contain {}",
                 root, cookie_dir
             )));
         }
-        setup_cookie_dir(cookie_dir)?;
-        let (sender, _) = broadcast::channel(1024);
+
+        let (sender_tx, sender_rx) = OptionalWatch::new();
```

Comment:
Why is this an `OptionalWatch`? It looks like we set it then immediately await it. Do we ever unset it? Change it?

Reply:
probably a case of YAGNI, removing